src/auth/user-record.ts

```diff
@@ -57,19 +58,226 @@ export interface CreateRequest extends UpdateRequest {
   uid?: string;
 }
 
+export interface Metadata {
+  createdAt?: string;
+  lastLoginAt?: string;
+}
+
+export interface MfaInfo {
+  mfaEnrollmentId: string;
+  displayName?: string;
+  phoneInfo?: string;
+  enrolledAt?: string;
+  [key: string]: any;
+}
+
+export interface ProviderUserInfo {
+  rawId: string;
+  displayName?: string;
+  email?: string;
+  photoUrl?: string;
+  phoneNumber?: string;
+  providerId: string;
+  federatedId?: string;
+}
+
+export interface Mfa {
+  mfaInfo?: MfaInfo[];
+}
+
+export interface GetAccountInfoUserResponse extends Metadata, Mfa {
+  localId: string;
+  email?: string;
+  emailVerified?: boolean;
+  phoneNumber?: string;
+  displayName?: string;
+  photoUrl?: string;
+  disabled?: boolean;
+  passwordHash?: string;
+  salt?: string;
+  customAttributes?: string;
+  validSince?: string;
+  tenantId?: string;
+  providerUserInfo?: ProviderUserInfo[];
+  [key: string]: any;
+}
+
+/** Enums for multi-factor identifiers. */
+export enum MultiFactorId {
+  Phone = 'phone',
+}
+
+/**
+ * Abstract class representing a multi-factor info interface.
+ */
+export abstract class MultiFactorInfo {
```

Comment:
Just `AuthFactor`? Subclass can be called `PhoneAuthFactor` or even just `PhoneFactor`.

Reply:
This is the name agreed upon in the API proposal and already implemented in the client SDKs which is in alpha. It can't be changed anymore.